internal/provider/resource_basistheory_application.go

```diff
@@ -125,9 +131,35 @@ func resourceBasisTheoryApplication() *schema.Resource {
 				Computed:    true,
 			},
 		},
+		SchemaVersion: 1,
+		StateUpgraders: []schema.StateUpgrader{
+			{
+				Type:    applicationInstanceResourceV0().CoreConfigSchema().ImpliedType(),
+				Upgrade: applicationInstanceStateUpgradeV0,
+				Version: 0,
+			},
+		},
+	}
+}
+
+func applicationInstanceResourceV0() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"create_key": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
 	}
 }
 
+func applicationInstanceStateUpgradeV0(_ context.Context, rawState map[string]any, _ any) (map[string]any, error) {
+	rawState["create_key"] = "false"
```

Comment:
I'm probably misunderstanding this, but this seems backwards to me - it sounded like this upgrades state from old version to current version, and pre-existing applications had their default keys auto created, so wouldn't the state be equivalent to an application now with `create_key = true`?

Reply:
you're right. I got this backwards